src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java

```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         }
         return sb.toString();
     }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     * and will verify that they are of correct rank to do so.
+     * @param player the players/senders name.
+     * @param theCommand which command to check if the player can use it.
+     * @param colonyId the id of the colony.
+     * @return boolean
+     */
+
+    public boolean canPlayerUseCommand(final EntityPlayer player, final Commands theCommand, final int colonyId)
+    {
+        if (isPlayerOpped(player, theCommand.toString()))
+        {
+            return true;
+        }
+
+        final Colony chkColony = ColonyManager.getColony(colonyId);
+        if(chkColony == null)
+        {
+            return false;
+        }
+        return canCommandSenderUseCommand(theCommand)
+                && canRankUseCommand(chkColony, player);
+    }
+
+    /**
+     * Checks if the player has the permission to use the command.
+     * By default officer and owner, overwrite this if other required.
+     * @param colony the colony.
+     * @param player the player.
+     * @return true if so.
+     */
+    public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final EntityPlayer player)
+    {
+        return colony.getPermissions().getRank(player).equals(Permissions.Rank.OFFICER) || colony.getPermissions().getRank(player).equals(Permissions.Rank.OWNER);
+    }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     *
+     * @param theCommand which command to check if the player can use it
+     * @return boolean
+     */
+    public boolean canCommandSenderUseCommand(Commands theCommand)
+    {
+        switch (theCommand)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Complete cases by adding the missing enum constants or add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)